### ISSUE TITLE
Current cover page generation logic loads whole bitstream

### ISSUE THREAD

**Issue by corrad82-4s** [DSpace/DSpace#8397]
**Describe the bug**
Non properly a bug, but a scenario we think could be worth some further comments. 
During pdf cover page generation, original bitstream is loaded in memory and then cover pdf page is attached. 
In case of large-sized files or peak of requests of files with cover pages, the usage of the in memory load could lead to performance issues. 
It could be worth seek a solution avoiding full load of file, using, for example a temporary file to generate the final file with cover. Maybe a different usage of method org.apache.pdfbox.pdmodel.PDDocument#load, invoked with proper MemoryUsageSettings could help in preventing problems.

The need of having the full file size returned by method org.dspace.disseminate.CitationDocumentServiceImpl#makeCitedDocument should be taken into account. Despite of the solution used, the correct size of file with cover must be returned as result.

**Comment by tdonohue** [comment]
Agreed this is a bug. Flagging as low priority for now though, as I'm not sure how many people even use this feature (it's recommended to leave it disabled for SEO purposes)

**Comment by Pranavv361** [comment]
Hi Tim and  corrad82-4s, me and my team are trying to automate cover page for University of Oklahoma Libraries and Open Initiatives for Scholarly Communication, Is it possible to customize(for ex: adding University logo) the available citation cover page - https://wiki.lyrasis.org/display/DSDOC7x/PDF+Citation+Cover+Page.
We currently use Dspace v6 and are planning to migrate from v6 to v7 in summer 2023.
If not, is it possible to add more metadata fields to the this cover page?
Any help is appreciated.
Thank you.

**Comment by saschaszott** [comment]
Another approach to merge the cover page with the existing bitstream PDF is using the `PDFMergerUtility` class of PDFBox (https://javadoc.io/doc/org.apache.pdfbox/pdfbox/latest/index.html).
